### DISCUSSION TITLE
Get current `html` source of an hs.webview object?

### DISCUSSION THREAD

**Discussion by luckman212** [Hammerspoon/hammerspoon#3075] [Q&A]
I'm trying to make a "scratchpad" type function in Hammerspoon, and have a skeleton working (see [here](https://github.com/Hammerspoon/hammerspoon/issues/3073))

What I want to be able to do is save/restore the contents of this `<textarea>` -- but the `hs.webview:html()` method only sets, it doesn't seem to get.

Anyone know a way around this?

**Comment by latenitefilms** [comment]
Use JavaScript to communicate between your web content and Lua:

http://www.hammerspoon.org/docs/hs.webview.usercontent.html

**Comment by luckman212** [comment]
@latenitefilms Any small examples of this that you know of? I'm not sure where to begin. Sorry.

**Reply by latenitefilms** [reply]
See:

https://github.com/CommandPost/CommandPost/blob/17bfb247ecee2d6204bbc89a5af83f729d739f56/src/plugins/core/preferences/manager/init.lua#L315

https://github.com/CommandPost/CommandPost/blob/17bfb247ecee2d6204bbc89a5af83f729d739f56/src/plugins/core/preferences/manager/html/panels.html#L18

Basically you can use JavaScript to send messages back to Lua.

**Reply by luckman212** [reply]
Thanks @latenitefilms it might still be over my head, but I'll try!

**Comment by luckman212** [comment]
@latenitefilms Took me the better part of the day but -- got it working! Thanks a million for the nudge in the right direction.

**Comment by luckman212** [comment]
To anyone stumbling onto this randomly, [this gist](https://gist.github.com/heptal/0103e664cbdd7b552f7444503b8022b4) also helped get me to the finish line.
